lib/tasks/wikirate.rake

```diff
@@ -89,13 +100,13 @@ namespace :wikirate do
         ActiveRecord::Base.connection.execute 'delete from sessions'
         
         # delete companies
-        ActiveRecord::Base.connection.execute "delete from cards where type_id = '651' and id not in ( #{company_ids[0...-1]} )"
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '#{Card::WikirateCompanyID}' and id not in ( #{company_ids[0...-1]} )"
         # delete topics
-        ActiveRecord::Base.connection.execute "delete from cards where type_id = '1010' and id not in ( #{topic_ids[0...-1]} )"
+        ActiveRecord::Base.connection.execute "delete from cards where type_id = '#{Card::WikirateTopicID}' and id not in ( #{topic_ids[0...-1]} )"
 
         # delete all webpage++link
-        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in ('631','651','699','974','1010','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
-        ActiveRecord::Base.connection.execute "delete from cards where type_id in ('631','699','974','1109','1591','1638','1690','2207','2317','2327','2754','2755','2813','2995','4010','4030') and id not in ( #{card_to_be_kept[0...-1]} )"
+        ActiveRecord::Base.connection.execute "delete ca from cards ca inner join cards le ON ca.left_id = le.id where le.type_id in (#{type_ids[0...-1]}) and ca.id not in  ( #{card_to_be_kept[0...-1]} )" 
+        ActiveRecord::Base.connection.execute "delete from cards where type_id in (#{type_ids_without_company_and_topic[0...-1]}) and id not in ( #{card_to_be_kept[0...-1]} )"
```

Comment:
I guess the `[0..-1]` is there to get rid of the comma at the end? -1 includes the end of the string so it should say `[0..-2]` or maybe easier to read: collect the ids in an array and use `join ','`. Then you get rid of the initialization statement as well `type_ids = %w{ ... }.map { ... }.join ','`